crates/sargon/src/signing/petition_types/general_role_with_hd_factor_instance.rs

```diff
@@ -0,0 +1,229 @@
+use crate::prelude::*;
+
+decl_role_with_factors!(
+    /// A general depiction of each of the roles in a `MatrixOfFactorInstances`.
+    /// `SignaturesCollector` can work on any `RoleKind` when dealing with a securified entity.
+    General,
+    HierarchicalDeterministicFactorInstance
+);
+
+impl TryFrom<(MatrixOfFactorInstances, RoleKind)>
+    for GeneralRoleWithHierarchicalDeterministicFactorInstances
+{
+    type Error = CommonError;
+
+    fn try_from(value: (MatrixOfFactorInstances, RoleKind)) -> Result<Self> {
+        let matrix = value.0;
+
+        let (threshold_factors, threshold, override_factors) = match value.1 {
+            RoleKind::Primary => (
+                matrix.primary_role.threshold_factors,
+                matrix.primary_role.threshold,
+                matrix.primary_role.override_factors,
+            ),
+            RoleKind::Recovery => (
+                matrix.recovery_role.threshold_factors,
+                matrix.recovery_role.threshold,
+                matrix.recovery_role.override_factors,
+            ),
+            RoleKind::Confirmation => (
+                matrix.confirmation_role.threshold_factors,
+                matrix.confirmation_role.threshold,
+                matrix.confirmation_role.override_factors,
+            ),
+        };
+
+        let general_role = GeneralRoleWithHierarchicalDeterministicFactorInstances::new(
+            threshold_factors
+                .iter()
+                .map(|f| HierarchicalDeterministicFactorInstance::try_from_factor_instance(f.clone()))
+                .collect::<Result<Vec<HierarchicalDeterministicFactorInstance>>>()?,
+            threshold,
+            override_factors
+                .iter()
+                .map(|f| HierarchicalDeterministicFactorInstance::try_from_factor_instance(f.clone()))
+                .collect::<Result<Vec<HierarchicalDeterministicFactorInstance>>>()?,
+        );
+
+        Ok(general_role)
+    }
+}
+
+impl From<GeneralRoleWithHierarchicalDeterministicFactorInstances>
```

Comment:
is this from used? it is a bit dangerous, it hides the fact that it uses the same role for ALL roles. 

So we should remove this. Or if it is only used inside unit tests we should put it behind `cfg(test)` **and** we should rename it, it should not be an `impl From` but rather a (named) constructor on `MatrixOfFactorInstances`, e.g. `fn using_general_role_for_all_roles()`.

Reply:
If this is used in non-unit tests but actual code, let me know and we need to rethink that code.